extensions/elytron-security-oauth2/runtime/src/main/java/io/quarkus/elytron/security/oauth2/runtime/OAuth2Config.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.elytron.security.oauth2.runtime;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * See http://docs.wildfly.org/14/WildFly_Elytron_Security.html#validating-oauth2-bearer-tokens
+ */
+@ConfigRoot(name = "oauth2", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class OAuth2Config {
+    /**
+     * The realm name
+     */
+    @ConfigItem(defaultValue = "Oauth2")
+    public String realmName;
+
+    /**
+     * The authentication mechanism
+     * TODO why is it configurable as it's hardcoded in the SecurityTemplate ?
+     */
+    @ConfigItem(defaultValue = "BEARER_TOKEN")
```

Comment:
Not sure it needs to be configurable at this point of time, do the alternatives exist ?

Reply:
I copied from the existing elytron-security extention but I don't see why it should be configurable. I'll remove it (and the auth mechanism name). We could make them configurable later if we want to.

Reply:
Agreed, for the moment only the bits which are required for the introspection call to succeed are needed

Reply:
removed